packages/fiori/src/BarcodeScannerDialog.ts

```diff
@@ -230,19 +235,64 @@ class BarcodeScannerDialog extends UI5Element {
 
 	_decodeFromCamera() {
 		const videoElement = this._getVideoElement();
-		this._codeReader.decodeFromVideoDevice(null, videoElement, (result: Result, err?: Exception) => {
-			this.loading = false;
-			if (result) {
-				this.fireEvent<BarcodeScannerDialogScanSuccessEventDetail>("scan-success",
-					{
-						text: result.getText(),
-						rawBytes: result.getRawBytes(),
-					});
-			}
-			if (err && !(err instanceof NotFoundException)) {
-				this.fireEvent<BarcodeScannerDialogScanErrorEventDetail>("scan-error", { message: err.message });
-			}
-		}).catch((err: Error) => this.fireEvent<BarcodeScannerDialogScanErrorEventDetail>("scan-error", { message: err.message }));
+
+		try {
+			this._codeReader.decodeFromVideoDevice(null, videoElement, (result: Result, err?: Exception) => {
+				this.loading = false;
+
+				if (result) {
+					this._handleDecodeSuccess(result, videoElement);
+				} else if (err && !(err instanceof NotFoundException)) {
+					this._handleDecodeError(err);
+				}
+			});
+		} catch (err) {
+			this._handleDecodeError(err);
+		}
+	}
+
+	_isWithinCenteredRegion(point: ResultPoint, scaleX: number, scaleY: number, videoRect: DOMRect, captureRegionRect: DOMRect) {
```

Comment:
This function does not use `this`, therefore should not be on the prototype. Move outside of the class to the module.